social_core/backends/oauth.py

```diff
@@ -40,8 +40,9 @@ def extra_data(self, user, uid, response, details=None, *args, **kwargs):
         """Return access_token and extra defined names to store in
         extra_data field"""
         data = super().extra_data(user, uid, response, details, *args, **kwargs)
-        data['access_token'] = response.get('access_token', '') or \
-            kwargs.get('access_token')
+        data['access_token'] = (response.get('access_token', '') or
+            kwargs.get('access_token') or
+            kwargs['request'].get('access_token') if 'request' in kwargs else None)
```

Comment:
This looks wrong. The request is already passed as named arg and should not appearing the kwargs

Reply:
From the debugging - args is actually an empty tuple. And kwargs is like this:
```
{'auth': <social_core.backends.vk.VKAppOAuth2 object at 0x7fd18e08dc40>,
 'is_new': False,
 'new_association': False,
 'pipeline_index': 8,
 'request': {'access_token': '...',
             ...},
 'social': <UserSocialAuth: John Doe>,
 'storage': <class 'social_django.models.DjangoStorage'>,
 'strategy': <social_django.strategy.DjangoStrategy object at 0x7fd18e0308b0>,
 'username': 'john'}
```

Reply:
Sorry, I misread the method signature...

Anyway looking again at this (and hopefully more carefully) - `OAuthAuth.extra_data` is definitely not the place to address this behavior. The `request` is here present only because `VKAppOAuth2.auth_complete` behaves differently than in the base OAuth2 class.

The proper fix would be to make `VKAppOAuth2.auth_complete` consistent with other OAuth2 backends.

Reply:
Sorry, yes, that makes sense. I made a change in `VKAppOAuth2.auth_complete`.